src/plugins/fs/generic.h

```diff
@@ -10,6 +10,22 @@ gchar* bd_fs_get_fstype (const gchar *device,  GError **error);
 gboolean bd_fs_freeze (const gchar *mountpoint, GError **error);
 gboolean bd_fs_unfreeze (const gchar *mountpoint, GError **error);
 
+typedef enum {
+    BD_FS_MKFS_LABEL     = 1 << 0,
+    BD_FS_MKFS_UUID      = 1 << 1,
+    BD_FS_MKFS_DRY_RUN   = 1 << 2,
+    BD_FS_MKFS_NODISCARD = 1 << 3,
+} BDFsMkfsOptionsFlags;
+
+typedef struct BDFsMkfsOptions {
+    gchar *label;
+    gchar *uuid;
+    gboolean dry_run;
+    gboolean no_discard;
+} BDFsMkfsOptions;
```

Comment:
Just thinking in terms of API stability for the future... would it make sense to add other commonly used filesystem _features_, no matter if they're going to be actually used or not? Extending the `struct` in the future would be difficult.

Thinking about `mkfs`-time options like `xattrs`, `compression`, `encryption`, `zoned` (SMR HDD, NVMe ZNS), `case-insensitive`, etc.

Reply:
I'm not sure about this. We could add some unused "reserve" to the struct that would allow us to add new features without changing the struct size.

Reply:
Adding some reserve would be an option. Listing all the possible options in advance is impossible, OTOH, I think.